### PR TITLE
Registry cloudfront middleware described

### PR DESCRIPTION
**Pull request by miminar**

### CODE REVIEW

**Comment by miminar** [comment]
@ahardin-rh a sensitive eye to written english is welcome here.

File: install_config/registry/extended_registry_configuration.adoc [hunk 1]
```diff
@@ -385,12 +385,78 @@ as well as a risk of serving image layer to user, who is no longer authorized
 to access it.
 ====
 
+[[middleware-cloudfront]]
+==== CloudFront middleware
```
Comment (ahardin-rh) [inline install_config/registry/extended_registry_configuration.adoc]: s/middleware/Middleware

File: install_config/registry/extended_registry_configuration.adoc [hunk 2]
```diff
@@ -385,12 +385,78 @@ as well as a risk of serving image layer to user, who is no longer authorized
...
 The link:https://docs.docker.com/registry/configuration/#cloudfront[*CloudFront*
 middleware extension] can be added to support AWS, CloudFront CDN storage
 provider. CloudFront middleware speeds up distribution of image content
 internationally. The blobs are distributed to several edge locations around the
 world. The client is always directed to the edge with the lowest latency.
 
+[NOTE]
+====
+The link:https://docs.docker.com/registry/configuration/#cloudfront[*CloudFront*
+middleware extension] can be used only together with
```
Comment (ahardin-rh) [inline install_config/registry/extended_registry_configuration.adoc]: s/can be used only together with/can be only used with

File: install_config/registry/extended_registry_configuration.adoc [hunk 3]
```diff
@@ -385,12 +385,78 @@ as well as a risk of serving image layer to user, who is no longer authorized
...
+link:https://docs.docker.com/registry/storage-drivers/s3/[S3] storage.
+It is utilized only during blob serving. Therefore, only blob downloads can be
+speeded up, not uploads.
```
Comment (ahardin-rh) [inline install_config/registry/extended_registry_configuration.adoc]: Consider rewriting last sentence:
Therefore, only blob downloads can be sped up; uploads can not be.

File: install_config/registry/extended_registry_configuration.adoc [hunk 4]
```diff
@@ -385,12 +385,78 @@ as well as a risk of serving image layer to user, who is no longer authorized
...
+        baseurl: https://jrpbyn0k5k88bi.cloudfront.net/ <3>
+        privatekey: /etc/docker/cloudfront-ABCEDFGHIJKLMNOPQRST.pem <4>
+        keypairid: ABCEDFGHIJKLMNOPQRST <5>
+    - name: openshift
+----
+<1> The S3 storage needs to be configured the same way regardless of CloudFront
+middleware.
+<2> The CloudFront storage middleware needs to be listed before OpenShift
+middleware.
+<3> The CloudFront base URL. In AWS management console, this is listed as
```
Comment (ahardin-rh) [inline install_config/registry/extended_registry_configuration.adoc]: s/AWS management console/the AWS management console

File: install_config/registry/extended_registry_configuration.adoc [hunk 5]
```diff
@@ -385,12 +385,78 @@ as well as a risk of serving image layer to user, who is no longer authorized
...
+documentation] on creating CloudFront key pairs for your trusted signers. The
+file needs to be mounted as a secret
+xref:../../dev_guide/secrets.adoc#dev-guide-secrets[secret] into the registry
+pod.
+<5> The ID of your Cloudfront key pair.
+====
+
+[[middleware-overriding]]
+==== Overriding middleware configuration options
 The *middleware* section may not be overridden using environment variables.
```
Comment (ahardin-rh) [inline install_config/registry/extended_registry_configuration.adoc]: s/may not/must not

File: install_config/registry/extended_registry_configuration.adoc [hunk 6]
```diff
@@ -385,12 +385,78 @@ as well as a risk of serving image layer to user, who is no longer authorized
...
 There are a few exceptions, however. For example:
 ====
```
Comment (ahardin-rh) [inline install_config/registry/extended_registry_configuration.adoc]: Add a blank line before the example begins

**Comment by ahardin-rh** [comment]
@miminar Just a few minor comments from me. Overall, looks good. Is this for 3.4? Thanks!

**Comment by miminar** [comment]
@ahardin-rh This is applicable to all the branches from 3.2 (including). Thanks for the review - addressed.

**Comment by ahardin-rh** [comment]
Thank you!

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../install_config/registry/extended_registry_configuration.adoc#install-config-registry-extended-configuration[Setting up the Registry -> Extended Registry Configuration]
|Described the xref:../install_config/registry/extended_registry_configuration.adoc#middleware-cloudfront[CloudFront middleware extension].
%

**Comment by adellape** [comment]
@ahardin-rh I'm removing the 3.2 label here because the registy topic hadn't been turned into a subdirectory of multiple topics yet (happened in 3.3). Can you apply the needed changes for 3.2 in their appropriate spot on that branch?

**Pull request merged**